crates/ott-common/src/discovery/dns.rs

```diff
@@ -107,4 +123,18 @@ mod test {
 
         assert_eq!(config.dns_server, Some(([127, 0, 0, 1], 53).into()));
     }
+
+    #[test]
+    fn dns_server_is_optional() {
+        let json = json!({
+            "service_port": 8080,
+            "dns_server": "",
```

Comment:
This is the case I'm more concerned about. Here you are still providing `dns_server`. This change will verify `dns_server` is optional.

```suggestion
```